sdk/python/feast/registry.py

```diff
@@ -69,54 +67,62 @@ def __init__(self, registry_path: str, repo_path: Path, cache_ttl: timedelta):
             raise Exception(
                 f"Registry path {registry_path} has unsupported scheme {uri.scheme}. Supported schemes are file and gs."
             )
-        self.cached_registry_proto_ttl = cache_ttl
+        # Read cache from the RegistryStore.
+        self.cached_registry_proto = self._get_registry_proto()
         return
 
     def _initialize_registry(self):
-        """Explicitly forces the initialization of a registry"""
-        self._registry_store.update_registry_proto(updater=None)
+        """Explicitly initializes the registry with an empty proto."""
+        self.cached_registry_proto = RegistryProto()
+        self.cached_registry_proto.registry_schema_version = REGISTRY_SCHEMA_VERSION
+        self._registry_store.update_registry_proto(self.cached_registry_proto)
```

Comment:
Hmm does this mean that it will write an empty registry object as soon as it's initialized? I guess that's okay but I wonder if it can be deferred to a `flush` call.

Reply:
Good point, I think we can defer it to `flush`.